src/USER-PLUMED/fix_plumed.cpp

```diff
@@ -0,0 +1,436 @@
+// LAMMPS units si :: Joule, m; s
+      energyUnits=0.001;
+      lengthUnits=1.e-9;
+      timeUnits=1.e-12;
+    } else if (force->boltz == 1.3806504e-16){
+// LAMMPS units cgs :: erg; cms;, s
+      energyUnits=6.0221418e13;
+      lengthUnits=1.e-7;
+      timeUnits=1.e-12;
+    } else if (force->boltz == 3.16681534e-6){
+// LAMMPS units electron :: Hartree, bohr, fs
+      energyUnits=2625.5257;
+      lengthUnits=0.052917725;
+      timeUnits=0.001;
+    } else error->all(FLERR,"Odd LAMMPS units, plumed cannot work with that");
+    p->cmd("setMDEnergyUnits",&energyUnits);
+    p->cmd("setMDLengthUnits",&lengthUnits);
+    p->cmd("setMDTimeUnits",&timeUnits);
+  }
+
+// Read fix parameters:
+  int next=0;
+  for(int i=3;i<narg;++i){
+    if(!strcmp(arg[i],"outfile")) next=1;
+    else if(next==1){
+      if(universe->existflag == 1){
+        // Each replica writes an independent log file
+        //  with suffix equal to the replica id
+        char str_num[32], logFile[1024];
+        sprintf(str_num,".%d",universe->iworld);
+        strncpy(logFile,arg[i],1024-32);
+        strcat(logFile,str_num);
+        p->cmd("setLogFile",logFile);
+        next=0;
+      } else {
+        // partition option not used
+        p->cmd("setLogFile",arg[i]);
+        next=0;
+      }
+    }
+    else if(!strcmp(arg[i],"plumedfile"))next=2;
+    else if(next==2){
+      p->cmd("setPlumedDat",arg[i]);
+      next=0;
+    }
+    else error->all(FLERR,"syntax error in fix plumed - use 'fix name plumed plumedfile plumed.dat outfile plumed.out' ");
+  }
+  if(next==1) error->all(FLERR,"missing argument for outfile option");
+  if(next==2) error->all(FLERR,"missing argument for plumedfile option");
+
+  p->cmd("setMDEngine","LAMMPS");
+
+  int natoms=int(atom->natoms);
+  p->cmd("setNatoms",&natoms);
+
+  double dt=update->dt;
+  p->cmd("setTimestep",&dt);
+
+  virial_flag=1;
+  thermo_virial=1;
+  scalar_flag = 1;
+
+// This is the real initialization:
+  p->cmd("init");
+
+// Define compute to calculate potential energy
+  id_pe = new char[7];
+  id_pe = (char *) "plmd_pe";
+  char **newarg = new char*[3];
+  newarg[0] = id_pe;
+  newarg[1] = (char *) "all";
+  newarg[2] = (char *) "pe";
+  modify->add_compute(3,newarg);
+  delete [] newarg;
+  int ipe = modify->find_compute(id_pe);
+  c_pe = modify->compute[ipe];
+
+// Define compute to calculate pressure tensor
+  id_press = new char[9];
+  id_press = (char *) "plmd_press";
+  newarg = new char*[5];
+  newarg[0] = id_press;
+  newarg[1] = (char *) "all";
+  newarg[2] = (char *) "pressure";
+  newarg[3] = (char *) "NULL";
+  newarg[4] = (char *) "virial";
+  modify->add_compute(5,newarg);
+  delete [] newarg;
+  int ipress = modify->find_compute(id_press);
+  c_press = modify->compute[ipress];
+ 
+  // Check if nh type fixes have been called
+  // See comment in the setup method
+  for (int i = 0; i < modify->nfix; i++) {
+    if ( strncmp(modify->fix[i]->style,"nph",3) ||
```

Comment:
please note, that there are more NH fixes, e.g. as part of the family of rigid integrators.

Reply:
This code was broken anyway and I've updated and corrected it. It now catches all fixes that maintain their own pressure compute.